### PR TITLE
Relation document candidate

### LINKED ISSUE #36: Relation Document-Candidate
Create relation between Document and Candidate

### PR DESCRIPTION
**Pull request by mikiribe14**
Defined the relationship between Document and Candidate

### CODE REVIEW

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/CreateDocumentStepDefs.java [hunk 1]
```diff
@@ -14,17 +16,23 @@
...
         this.documentRepository = documentRepository;
         this.selectionProcessRepository = selectionProcessRepository;
+        this.candidateRepository = candidateRepository;
+
     }
     @When("I create a document with title {string}")
     public void iCreateANewDocumentWithTitle(String title) throws Throwable {
         Document document = new Document();
         document.setTitle(title);
+        documentRepository.save(document);
```
Comment (bergacat1) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/CreateDocumentStepDefs.java]: If you store the Document here, you can't check if the Document is correctly created through the POST you do below.

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/CreateDocumentStepDefs.java [hunk 2]
```diff
@@ -66,4 +74,42 @@ public void itHasBeenCreatedADocumentWithTitleAndVacancy(String vacancy) throws
...
+    @When("I assign the candidate named {string} to the document titled {string}")
+    public void iAssignCandidateToDocument(String cand_name, String doc_title) throws Throwable {
+        Candidate candidate = new Candidate();
+        candidate.setName(cand_name);
+        candidateRepository.save(candidate);
+
+        Document document = new Document();
+        document.setTitle(doc_title);
+        document.setCandidate(candidate);
+        documentRepository.save(document);
```
Comment (bergacat1) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/CreateDocumentStepDefs.java]: Again, you shouldn't save the Document here.

File: src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateDocumentStepDefs.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        this.stepDefs = stepDefs;
+        this.documentRepository = documentRepository;
+    }
+
+    @When("I change the title of the document with id {string} to {string}")
+    public void iChangeTitleOfDocumentTo(String id, String title) throws Throwable {
+
+        Document doc = documentRepository.findById(Long.valueOf(id)).get();
+        doc.setTitle(title);
+        documentRepository.save(doc);
```
Comment (bergacat1) [inline src/test/java/cat/udl/eps/softarch/academicrecruit/steps/UpdateDocumentStepDefs.java]: Same here.

**Review by bergacat1 — DISMISSED** [review]

**Comment by mikiribe14** [comment]
I removed the storing of the Document in all tests. It should be fine now

**Review by rogargon — APPROVED** [review]

**Pull request merged**